Packages/install_redeem

```diff
@@ -2,25 +2,26 @@
 # SPDX-License-Identifier: GPL-2.0
 
 set -e
-set -x
 for f in `ls ${VERSIONING}/*`
   do
     source $f
   done
 cd Packages
 
 echo "**install_redeem**"
+set -x
 if [ ! -d $REDEEM_HOME ]; then
   git clone $REDEEM_REPOSITORY --single-branch --branch $REDEEM_BRANCH $REDEEM_HOME
 fi
 cd $REDEEM_HOME
 git pull
 git checkout $REDEEM_BRANCH
-make install
+python2 setup.py develop
```

Comment:
I would prefer if we kept `make install` for production/end user purposes.
`python2 setup.py develop` should be meant for developer installations, not proper final packaging, imo.

Reply:
If the script is intended to generate a "final packaging", then I would agree IF you then remove all of the source and build. We can hide the python setup inside the Makefile. But I do want to get rid of the eggs being installed on systems that are running from git updates. The egg form is like a `deb` and is only a distribution mechanism. End users who update from source are not doing distributions. And we have run into a number of problems with the updates not really taking over because an older egg was left behind. By eliminating the build artifacts, we also save memory. And I still need to get rid of unused items so that we can run on the older BBBs that have only 2GB.

But where we distribute the source in the `git` tree, there is no advantage in doing the packaging. And keeping it "unpackaged" makes it quite a bit simpler for the user to upgrade from source.  IMHO, the alternative is to do the packaging on our development systems and "publish" the result (so the client just does a `pip install`

Reply:
Well I'd say that the `make install` script should take care of cleanup of the previous versions' eggs. When you do `apt-get upgrade` it removes previous versions of firefox to install the new one. Why wouldn't we do the same in this case? And while I agree we don't need the egg files in general, it indicates the build system needs to be updated (i.e. make install needs to be fixed), but it's a lot more intuitive for users to type `make install` than `python setup.py develop`. Even `python setup.py install` would be better from a user perspective...

Reply:
I'm happy with hiding the `python setup`, or `pip ...` or `apt-get .. ` in a `Makefile` and doing `make install`. But, perhaps implementing `make upgrade` would be more appropriate. I still argue that we should not be building "distribution packaging" on the client machine. When you do `apt-get upgrade` it doesn't build from source. It downloads a package that was generated by the distributor.

Reply:
I can also argue that this entire collection of scripts is the "Makefile" for our Umikaze distribution.
As such, hiding the internal build mechanisms in another layer doesn't really gain us much unless we standardize in a language independent way the internal method that we perform the install step.

Reply:
My main complaint isn't about this particular script. It's about the fact that we seem to have only one reliable way to install Redeem that isn't standard.

As noted - I'd prefer a `make install` or even `make upgrade` that works reliably for end-users doing an upgrade of Redeem, even if it isn't the mechanism used here. I just highlighted it here because this is where it's obvious that the installation method is incomplete.